ignek_intranet/modules/employee-web/src/main/java/com/ignek/intranet/employee/portlet/EmployeeWebPortlet.java

```diff
@@ -0,0 +1,153 @@
+		String addressLine2 = ParamUtil.getString(actionRequest, "addressLine2");
+		String city = ParamUtil.getString(actionRequest, "city");
+		long zipCode = ParamUtil.getLong(actionRequest, "zipCode");
+		String designation = ParamUtil.getString(actionRequest, "designation");
+		
+		employeeLocalService.addEmployee(empId, userUId, firstName, lastName, emailAddress, phoneNumber, addressLine1, addressLine2, city, zipCode, designation);
+ 
+		employeeServiceImpl.addUser(userId,userUId, firstName, lastName,companyId,emailAddress);
+		
+	}
+	
+	
+	
+	
+	
+	@Override
+	public void render(RenderRequest renderRequest, RenderResponse renderResponse)
+			throws IOException, PortletException {
+		
+		
+		List<Employee> employeeList = employeeLocalService.getEmployees(QueryUtil.ALL_POS,QueryUtil.ALL_POS);
+		renderRequest.setAttribute("employeeList", employeeList);
+		
+		super.render(renderRequest, renderResponse);
+	}	
+	
+
+	
+	@ProcessAction(name = "updateEmployee")
+	public void updateEmployee(ActionRequest actionRequest, ActionResponse actionResponse) throws PortalException {
+		ThemeDisplay themeDisplay = (ThemeDisplay) actionRequest.getAttribute(WebKeys.THEME_DISPLAY);
+		long empId = ParamUtil.getLong(actionRequest, "empId", GetterUtil.DEFAULT_LONG);
+		long empUserUId= employeeLocalService.fetchEmployee(empId).getUserId();
+		long userUId= userLocalService.getUser(empUserUId+1).getUserId();
+		long userId = themeDisplay.getUserId();
+		long companyId= themeDisplay.getCompanyId();
+		String firstName = ParamUtil.getString(actionRequest, "firstName", GetterUtil.DEFAULT_STRING);
+		String lastName = ParamUtil.getString(actionRequest, "lastName", GetterUtil.DEFAULT_STRING);
+		String emailAddress = ParamUtil.getString(actionRequest, "emailAddress", GetterUtil.DEFAULT_STRING);
+		long phoneNumber = ParamUtil.getLong(actionRequest, "phoneNumber", GetterUtil.DEFAULT_LONG);
+		String addressLine1 = ParamUtil.getString(actionRequest, "addressLine1", GetterUtil.DEFAULT_STRING);
+		String addressLine2 = ParamUtil.getString(actionRequest, "addressLine2", GetterUtil.DEFAULT_STRING);
+		String city = ParamUtil.getString(actionRequest, "city", GetterUtil.DEFAULT_STRING);
+		long zipCode = ParamUtil.getLong(actionRequest, "zipCode", GetterUtil.DEFAULT_LONG);
+		String designation = ParamUtil.getString(actionRequest, "designation", GetterUtil.DEFAULT_STRING);
+		
+		employeeLocalService.updateEmployee(empId, empUserUId, firstName, lastName, emailAddress, phoneNumber, addressLine1, addressLine2, city, zipCode, designation);
+		
+		employeeServiceImpl.updateUser(userId, userUId,firstName, lastName, companyId, emailAddress);
+		
+	}
+ 
+	
+	
+	@ProcessAction(name = "deleteEmployee")
+	public void deleteEmployee(ActionRequest actionRequest, ActionResponse actionResponse) {
+
+		long empId = ParamUtil.getLong(actionRequest, "empId", GetterUtil.DEFAULT_LONG);
+		long empUserUId= employeeLocalService.fetchEmployee(empId).getUserId();
```

Comment:
Add validations here. It may throw exceptions if employee data is not available for that empId.